client/src/utils/helpers.js

```diff
@@ -15,14 +15,16 @@ var _ = {
     }
   },
   eachAsync: function(obj, fn, cb){
-    var c = obj.length;
+    var callbacksLeft = obj.length;
     
-    var next = function(){
-      c--;
-      if(c === 0){ 
-        var a = cb;
+    var next = function(err){
+      callbacksLeft--;
+      if(callbacksLeft === 0 || !!err){
+
+        var cbCopy = cb;
```

Comment:
This 2 lines are to ensure main callback is called just once. So basically I am overriding main callback with noop (function(){}) so that future executions do nothing. This is mostly for handling the scenario of when I fire the callback as soon as the error happens (so, other things could complete later but I ignore them).